rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6551,6 +6553,318 @@ private void assertRefundInProcessPegInVersionLegacy(
         Assert.assertFalse(provider.getHeightIfBtcTxhashIsAlreadyProcessed(btcTx.getHash()).isPresent());
     }
 
+    private Address getFastBridgeAddressFromRedeemScript(Script redeemScript, Sha256Hash derivationArgumentHash) {
+        Script fastBridgeRedeemScript = FastBridgeRedeemScriptParser.createMultiSigFastBridgeRedeemScript(
+            redeemScript,
+            derivationArgumentHash
+        );
+
+        Script fastBridgeP2SH = ScriptBuilder.createP2SHOutputScript(fastBridgeRedeemScript);
+        return Address.fromP2SHScript(bridgeConstants.getBtcParams(), fastBridgeP2SH);
+    }
+
+    private BigInteger registerFastBridgeBtcTransaction_RSKIP293(
+            ActivationConfig.ForBlock activations,
+            Coin valueToSend,
+            boolean includeActiveFederation,
+            boolean includeRetiringFederation,
+            boolean retiringFederationExists) throws IOException, BlockStoreException, BridgeIllegalArgumentException {
+
+        Context btcContext = mock(Context.class);
+        when(btcContext.getParams()).thenReturn(bridgeConstants.getBtcParams());
+
+        Federation activeFederation = getFederation(bridgeConstants, "fa03", "fa04");
+        Federation retiringFederation = getFederation(bridgeConstants, "fa01", "fa02");
+
+        Repository repository = spy(createRepository());
+
+        BridgeStorageProvider provider = spy(new BridgeStorageProvider(repository, contractAddress, bridgeConstants, activations));
```

Comment:
Because it's being used for the method `validationsForRegisterBtcTransaction`. They used to be a mock as well as the method in the first version I pushed for review, but Jose suggested that I avoid mocking methods that are part of the method I'm testing.

Reply:
It's ok to not mock methods. What I mean is that you could mock Repository and BridgeStorageProvider instances. And set them to return the values you need for each call. That way it creates a more controlled environment in the test, and we ensure to be testing only one class while not depending on the implementation of other class instances.

Reply:
Done.